[miaopai] error while downloading from miaopai

annie -d http://www.miaopai.com/show/jgQhqCZaXT7ZBfkw2FXPXb1w2mvB3LFSh1CIIQ__.htm                                                                                                                                                             

annie: version 0.7.6, A fast, simple and clean video downloader.


URL:         http://www.miaopai.com/show/jgQhqCZaXT7ZBfkw2FXPXb1w2mvB3LFSh1CIIQ__.htm
Method:      GET
Headers:     http.Header{
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.146 Safari/537.36"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Referer":         {"http://www.miaopai.com/show/jgQhqCZaXT7ZBfkw2FXPXb1w2mvB3LFSh1CIIQ__.htm"},
}
Status Code: 200
2018/07/09 14:56:57 https://gslb.miaopai.com/stream/jgQhqCZaXT7ZBfkw2FXPXb1w2mvB3LFSh1CIIQ__.mp4?ssig=586d3363f82239df3331958dba0ed304&time_stamp=1531108612159&cookie_id=&vend=1&os=3&partner=1&platform=2&cookie_id=&refer=miaopai&scid=jgQhqCZaXT7ZBfkw2FXPXb1w2mvB3LFSh1CIIQ__
panic: Get https://txycdn.miaopai.com/stream/jgQhqCZaXT7ZBfkw2FXPXb1w2mvB3LFSh1CIIQ___16.mp4?ssig=1a58973e9cc5120fd8b2b0c59da71dad&time_stamp=1531108616828: x509: certificate is valid for *.cdn.myqcloud.com, *.2144.com, *.dpfile.com, *.duba.com, *.flash.2144.com, *.flash.cn, *.sogoucdn.com, *.zhaopin.cn, *.zhaopin.com, not txycdn.miaopai.com

goroutine 1 [running]:
github.com/iawia002/annie/request.Request(0x13236a7, 0x3, 0xc42026a274, 0xfe, 0x0, 0x0, 0xc4201cf968, 0xc4201fc460)
	/private/tmp/annie-20180708-4272-jsdnj9/annie-0.7.6/src/github.com/iawia002/annie/request/request.go:97 +0xb74
github.com/iawia002/annie/request.Headers(0xc42026a274, 0xfe, 0x7ffeefbffaf1, 0x48, 0x2)
	/private/tmp/annie-20180708-4272-jsdnj9/annie-0.7.6/src/github.com/iawia002/annie/request/request.go:149 +0x110
github.com/iawia002/annie/request.Size(0xc42026a274, 0xfe, 0x7ffeefbffaf1, 0x48, 0x1)
	/private/tmp/annie-20180708-4272-jsdnj9/annie-0.7.6/src/github.com/iawia002/annie/request/request.go:155 +0x4d
github.com/iawia002/annie/extractors.Miaopai(0x7ffeefbffaf1, 0x48, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/annie-20180708-4272-jsdnj9/annie-0.7.6/src/github.com/iawia002/annie/extractors/miaopai.go:17 +0x171
main.download(0x7ffeefbffaf1, 0x48)
	/private/tmp/annie-20180708-4272-jsdnj9/annie-0.7.6/src/github.com/iawia002/annie/main.go:84 +0x712
main.main()
	/private/tmp/annie-20180708-4272-jsdnj9/annie-0.7.6/src/github.com/iawia002/annie/main.go:129 +0xe0

--- a/extractors/miaopai_test.go
+++ b/extractors/miaopai_test.go
@@ -13,14 +13,14 @@ func TestMiaopai(t *testing.T) {
 		name string
 		args test.Args
 	}{
-		// {
-		// 	name: "normal test",
-		// 	args: test.Args{
-		// 		URL:   "https://www.miaopai.com/show/nPWJvdR4z2Bg1Sz3PJpNYffjpDgEiuv4msALgw__.htm",
-		// 		Title: "情人节特辑：一个来自绝地求生的爱情故事，送给已经离开的你-绝地求生大逃杀的秒拍",
-		// 		Size:  12135847,
-		// 	},
-		// },
+		{
+			name: "normal test",
+			args: test.Args{
+				URL:   "https://www.miaopai.com/show/nPWJvdR4z2Bg1Sz3PJpNYffjpDgEiuv4msALgw__.htm",
+				Title: "情人节特辑：一个来自绝地求生的爱情故事，送给已经离开的你-绝地求生大逃杀的秒拍",
+				Size:  12135847,
+			},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

--- a/extractors/weibo_test.go
+++ b/extractors/weibo_test.go
@@ -13,38 +13,38 @@ func TestWeibo(t *testing.T) {
 		name string
 		args test.Args
 	}{
-		// {
-		// 	name: "normal test",
-		// 	args: test.Args{
-		// 		URL:   "https://m.weibo.cn/2815133121/G9VBqbsWM",
-		// 		Title: "当你超过25岁再去夜店……",
-		// 		Size:  3112080,
-		// 	},
-		// },
-		// {
-		// 	name: "fid url test",
-		// 	args: test.Args{
-		// 		URL:   "https://weibo.com/tv/v/Ga7XazXze?fid=1034:4a65c6e343dc672789d3ba49c2463c6a",
-		// 		Title: "看完更加睡不着了[二哈]",
-		// 		Size:  438757,
-		// 	},
-		// },
-		// {
-		// 	name: "title test",
-		// 	args: test.Args{
-		// 		URL:   "https://m.weibo.cn/status/4237529215145705",
-		// 		Title: `近日，日本视错觉大师、明治大学特任教授\"杉原厚吉的“错觉箭头“作品又引起世界人民的关注。反射，透视和视角的巧妙结合产生了这种惊人的幻觉：箭头向右？转过来...`,
-		// 		Size:  1125984,
-		// 	},
-		// },
-		// {
-		// 	name: "weibo.com test",
-		// 	args: test.Args{
-		// 		URL:   "https://weibo.com/1642500775/GjbO5ByzE",
-		// 		Title: "让人怦然心动的小姐姐们 via@大懒糖",
-		// 		Size:  9198410,
-		// 	},
-		// },
+		{
+			name: "normal test",
+			args: test.Args{
+				URL:   "https://m.weibo.cn/2815133121/G9VBqbsWM",
+				Title: "当你超过25岁再去夜店……",
+				Size:  3112080,
+			},
+		},
+		{
+			name: "fid url test",
+			args: test.Args{
+				URL:   "https://weibo.com/tv/v/Ga7XazXze?fid=1034:4a65c6e343dc672789d3ba49c2463c6a",
+				Title: "看完更加睡不着了[二哈]",
+				Size:  438757,
+			},
+		},
+		{
+			name: "title test",
+			args: test.Args{
+				URL:   "https://m.weibo.cn/status/4237529215145705",
+				Title: `近日，日本视错觉大师、明治大学特任教授\"杉原厚吉的“错觉箭头“作品又引起世界人民的关注。反射，透视和视角的巧妙结合产生了这种惊人的幻觉：箭头向右？转过来...`,
+				Size:  1125984,
+			},
+		},
+		{
+			name: "weibo.com test",
+			args: test.Args{
+				URL:   "https://weibo.com/1642500775/GjbO5ByzE",
+				Title: "让人怦然心动的小姐姐们 via@大懒糖",
+				Size:  9198410,
+			},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

--- a/request/request.go
+++ b/request/request.go
@@ -3,6 +3,7 @@ package request
 import (
 	"compress/gzip"
 	"compress/zlib"
+	"crypto/tls"
 	"fmt"
 	"io"
 	"io/ioutil"
@@ -29,6 +30,7 @@ func Request(
 	transport := &http.Transport{
 		DisableCompression:  true,
 		TLSHandshakeTimeout: 10 * time.Second,
+		TLSClientConfig:     &tls.Config{InsecureSkipVerify: true},
 	}
 	if config.Proxy != "" {
 		var httpProxy, err = netURL.Parse(config.Proxy)

